hasher-matcher-actioner/hmalib/matchers/matchers_base.py

```diff
@@ -223,32 +224,18 @@ def get_index(self, signal_type: t.Type[SignalType]) -> SignalTypeIndex:
             self._cached_indexes[signal_type], index_cls
         ):
             with metrics.timer(metrics.names.indexer.download_index):
-                self._cached_indexes[signal_type] = index_cls.load(
-                    bucket_name=self.index_bucket_name
-                )
+                self._cached_indexes[signal_type] = self.index_store.load(index_cls)
 
         return self._cached_indexes[signal_type]
 
     @classmethod
     def _get_index_for_signal_type_matching(
         cls, signal_type: t.Type[SignalType], max_custom_threshold: int
     ):
-        indexes = signal_type.get_index_cls()
-        # disallow empty list
-        assert indexes
-        if len(indexes) == 1:
-            # if we only have one option just return
-            return indexes[0]
-
-        indexes.sort(key=lambda i: i.get_index_max_distance())
-
-        for index_cls in indexes:
-            if max_custom_threshold <= index_cls.get_index_max_distance():
-                return index_cls
-
-        # if we don't have an index that supports max threshold
-        # just return the one if the highest possible max distance
-        return indexes[-1]
+        # TODO: Figure out how to switch index type when max_custom_threshold
+        # changes.
```

Comment:
This and dynamically adjusting the SignalType configs are both hard problems at the moment (i.e. disabling signal types).